cmake/SetupCodeChecks.cmake

```diff
@@ -753,3 +786,94 @@ macro(blt_add_uncrustify_target)
     endif()
 
 endmacro(blt_add_uncrustify_target)
+
+
+##------------------------------------------------------------------------------
+## blt_add_yapf_target( NAME              <Created Target Name>
+##                      MODIFY_FILES      [TRUE | FALSE (default)]
+##                      CFG_FILE          <Yapf Configuration File>
+##                      PREPEND_FLAGS     <Additional Flags to Yapf>
+##                      APPEND_FLAGS      <Additional Flags to Yapf>
+##                      COMMENT           <Additional Comment for Target Invocation>
+##                      WORKING_DIRECTORY <Working Directory>
+##                      SRC_FILES         [FILE1 [FILE2 ...]] )
+##
+## Creates a new target with the given NAME for running yapf over the given SRC_FILES.
+##------------------------------------------------------------------------------
+macro(blt_add_yapf_target)
+
+    ## parse the arguments to the macro
+    set(options)
+    set(singleValueArgs NAME MODIFY_FILES CFG_FILE COMMENT WORKING_DIRECTORY)
+    set(multiValueArgs SRC_FILES PREPEND_FLAGS APPEND_FLAGS)
+
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # Check/Set required parameters
+    if(NOT DEFINED arg_NAME)
+        message(FATAL_ERROR "blt_add_yapf_target requires a NAME parameter")
+    endif()
+
+    if(NOT DEFINED arg_CFG_FILE)
+        message(FATAL_ERROR "blt_add_yapf_target requires a CFG_FILE parameter")
+    endif()
+
+    if(NOT DEFINED arg_SRC_FILES)
+        message(FATAL_ERROR "blt_add_yapf_target requires a SRC_FILES parameter")
+    endif()
+
+    if(NOT DEFINED arg_MODIFY_FILES)
+        set(arg_MODIFY_FILES FALSE)
+    endif()
+
+    if(DEFINED arg_WORKING_DIRECTORY)
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+
+    set(_generate_target TRUE)
+
+    # Check the version -- output is of the form "yapf X.Y.Z"
+    execute_process(
+        COMMAND ${YAPF_EXECUTABLE} --version
+        OUTPUT_VARIABLE _version_str
+        OUTPUT_STRIP_TRAILING_WHITESPACE )
+    string(REGEX MATCH "([0-9]+(\\.)?)+$" _yapf_version ${_version_str})
+
+    if(BLT_REQUIRED_YAPF_VERSION)
+        # The user may only specify a part of the version (e.g. just the maj ver)
+        # so check for substring
+        string(FIND ${_yapf_version} ${BLT_REQUIRED_YAPF_VERSION} VERSION_POS)
+        if (NOT VERSION_POS EQUAL 0)
```

Comment:
I found a bug in the code you based this on.  #413 will be merged soon and fixes this.

Reply:
Yes, I saw the behavior you observed in #413 while testing this PR. Thanks for the heads up!

Reply:
Added this change in amended commit [05e425c8](https://github.com/LLNL/blt/pull/412/commits/05e425c873feedc59e386f0c73ebebf68c072b03).

Reply:
Github is going insane.. I've now gotten 6 emails about his one comment.

Reply:
Can you guard this message (and update the message) like the other style checkers?  Otherwise it ends up printing this message once per target, which is a lot.

Reply:
Added in [dea10a35b](https://github.com/LLNL/blt/pull/412/commits/dea10a35b51385c501b17b8f97f6fde5022885e6). If I understand the message guard, a version mismatch would disable both the `style` and `check` targets?

Reply:
It disables the per-target, beautifier-level `style` and `check` targets not the higher level actual `style` and `check` targets. For example, if using the high level macro `blt_add_code_checks(PREFIX foo ...)`, `foo_yapf_style` and `foo_yapf_check` would not be created.